non_logged_in_area/views.py

```diff
@@ -26,7 +26,12 @@ def get_context_data(self, **kwargs):
         context['regions'] = Region.objects.annotate(
             facility_count=Count('areas__places__facilities')).exclude(
             facility_count=0).prefetch_related('areas', 'areas__region').all()
-        context['facilities'] = Facility.objects.select_related('place',
+        facilities = Facility.objects.select_related('place',
                                                                 'place__area',
                                                                 'place__area__region').order_by('place').all()
+        facilities_with_shifts = []
+        for i in facilities:
+            if len(i.open_shifts) > 0:
+                facilities_with_shifts.append(i)
+        context['facilities'] = facilities_with_shifts
```

Comment:
Why did you not apply another `.filter()` to the queryset?

Reply:
tried 3min but failed. Does it make a difference? Please correct :)